src/common/offline_transformations/src/pruning/propagate_masks.cpp

```diff
@@ -1420,6 +1432,187 @@ class ngraph::pass::mask_propagation::Transpose : public MatcherPass {
+                return false;
+
+            auto split_lengths = split_lengths_const->cast_vector<int64_t>();
+            auto axis = axis_node->cast_vector<int64_t>()[0];
+            if (axis < 0)
+                axis += input_mask->size();
+
+            // adjust split_lengths if needed
+            // split_lengths can contain -1 value
+            int minus_one_length_idx = -1;
+            int64_t total_lengths = 0;
+            for (int i = 0; i < split_lengths.size(); i++) {
+                if (split_lengths[i] == -1) {
+                    minus_one_length_idx = i;
+                    continue;
+                }
+                total_lengths += split_lengths[i];
+            }
+            if (minus_one_length_idx >= 0 && !input_mask->at(axis).empty()) {
+                const auto& input_shape = input.get_partial_shape();
+                if (input_shape[axis].is_dynamic())
+                    return false;
+                auto split_dim = input_shape[axis].get_length();
+                split_lengths[minus_one_length_idx] = split_dim - total_lengths;
+            }
+
+            uint64_t split_start = 0;
+            uint64_t split_end = 0;
+            std::vector<ngraph::Mask::Ptr> output_masks;
+            std::vector<ngraph::Mask*> output_masks_raw;
+            auto input_mask_raw = input_mask.get();
+            for (size_t i = 0; i < split->get_output_size(); i++) {
+                split_end += split_lengths[i];
+                auto output_mask = std::make_shared<ngraph::Mask>();
+                output_mask->copy_and_slice_mask_from(input_mask_raw, axis, split_start, split_end);
+                output_masks.push_back(output_mask);
+                output_masks_raw.push_back(output_mask.get());
+                output_mask->add_callback(
+                    [input_mask_raw, axis, split_start, split_end](ngraph::Mask::Ptr cur_mask) -> bool {
+                        cur_mask->copy_and_slice_mask_from(input_mask_raw, axis, split_start, split_end);
+                        return true;
+                    },
+                    input_mask);
+                split_start = split_end;
+                ngraph::setMask(split->output(i), output_masks[i]);
+            }
+
+            auto input_mask_callback = [output_masks_raw, axis, split_lengths](ngraph::Mask::Ptr cur_mask) -> bool {
+                cur_mask->clean_dim_values();
+                uint64_t split_start = 0;
+                for (size_t i = 0; i < output_masks_raw.size(); i++) {
+                    auto mask = output_masks_raw[i];
+                    for (size_t j = 0; j < mask->size(); j++) {
+                        const auto& dim_mask = mask->at(j);
+                        if (j == axis) {
+                            for (auto d : dim_mask)
+                                cur_mask->at(j).insert(d + split_start);
+                        } else if (!dim_mask.empty()) {
+                            cur_mask->at(j) = dim_mask;
```

Comment:
In case one of output nodes invalidates masks  for `dim != axis` (addition with 1, for example), so this information will be ignored by input mask of split op. Such logic looks invalid. Why input mask callback couldn't intersect all masks from output tensors for `forall dim, dim != axis` and then send callback back to output edges?